mypy/types.py

```diff
@@ -583,19 +583,39 @@ class ParamSpecType(TypeVarLikeType):
     prefix: 'Parameters'
 
     def __init__(
-         self, name: str, fullname: str, id: Union[TypeVarId, int], flavor: int,
-         upper_bound: Type, *, line: int = -1, column: int = -1,
-         prefix: Optional['Parameters'] = None
+         self, name: str, fullname: str, id: Union[TypeVarId, int], flavor: int, *,
+         upper_bound: Optional[Type] = None,
+         named_type_func: Optional[Callable[..., 'Instance']] = None, line: int = -1,
```

Comment:
I moved the code into `semanal_shared` -- although it could actually possibly be moved into `typeanal`, since it seems paramspec args and paramspec kwargs are only ever constructed in `typeanal` at the moment. (But perhaps that might change in the future -- what do you think?)